apps/voting/contracts/Voting.sol

```diff
@@ -515,12 +547,69 @@ contract Voting is IForwarder, AragonApp {
 
         vote_.executed = true;
 
+        address[] memory blacklist;
         bytes memory input = new bytes(0); // TODO: Consider input for voting scripts
-        runScript(vote_.executionScript, input, new address[](0));
 
+        if (vote_.disputableStatus == DisputableStatus.Invalid) {
+            blacklist = new address[](0);
+        } else {
+            blacklist = new address[](1);
+            blacklist[0] = address(_getAgreement());
+            _closeAction(vote_.actionId);
+            vote_.disputableStatus = DisputableStatus.Closed;
+        }
+
+        runScript(vote_.executionScript, input, blacklist);
         emit ExecuteVote(_voteId);
     }
 
+    /**
+    * @dev Challenge a vote
+    * @param _voteId Identification number of the vote to be challenged
+    */
+    function _onDisputableChallenged(uint256 _voteId, address /* _challenger */) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_canPause(vote_), ERROR_CANNOT_PAUSE_VOTE);
+
+        vote_.disputableStatus = DisputableStatus.Paused;
+        vote_.pausedAt = getTimestamp64();
+        emit PauseVote(_voteId);
+    }
+
+    /**
+    * @dev Allow a vote
+    * @param _voteId Identification number of the vote to be allowed
+    */
+    function _onDisputableAllowed(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_isPaused(vote_), ERROR_VOTE_NOT_PAUSED);
+
+        vote_.disputableStatus = DisputableStatus.Active;
+        vote_.pauseDuration = getTimestamp64().sub(vote_.pausedAt);
+        emit ResumeVote(_voteId);
+    }
+
+    /**
+    * @dev Reject a vote
+    * @param _voteId Identification number of the vote to be rejected
+    */
+    function _onDisputableRejected(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_isPaused(vote_), ERROR_VOTE_NOT_PAUSED);
+
+        vote_.disputableStatus = DisputableStatus.Cancelled;
+        vote_.pauseDuration = getTimestamp64().sub(vote_.pausedAt);
+        emit CancelVote(_voteId);
+    }
+
+    /**
+    * @dev Void an entry
+    * @param _voteId Identification number of the entry to be voided
+    */
+    function _onDisputableVoided(uint256 _voteId) internal {
+        _onDisputableRejected(_voteId);
```

Comment:
Are we sure this is what we want? I wonder if this should be an init param (to decide the behavior, I mean)

Reply:
Based on a conversation with @lkngtn that's the case. However, we could parameterize this in the future